### PR TITLE
STYL: replace autotyping with ruff's ANN autofixes

### PR DESCRIPTION
**Pull request by twoertwein**
ruff's ANN has many of the autofixes from autotyping but it doesn't have all of them. Ruff doesn't annotate parameters with simple default values (autotyping would change `def foo(x=1)` to `def foo(x: int = 1)`).

### CODE REVIEW

File: .pre-commit-config.yaml
Comment (mroeschke) [inline .pre-commit-config.yaml:30]: Could we just include `ANN` as a rule in pyproject.toml?

Comment (twoertwein) [reply .pre-commit-config.yaml:30]: I wish, but there are way too many violations (ANN wants everything to be annotated) and only some of them have in some cases an autofix. I would like only to use the autofixes. If there is a better way to achieve that, I'm all for that - but I didn't find one.

Comment (mroeschke) [reply .pre-commit-config.yaml:30]: Hmm since it appears `ANN204` is the only auto-fix rule so far, could we just include that code?

Comment (twoertwein) [reply .pre-commit-config.yaml:30]: Let me put this back to draft and wait until #54271 is merged

Comment (twoertwein) [reply .pre-commit-config.yaml:30]: There are 191 methods without return annotations.

Comment (twoertwein) [reply .pre-commit-config.yaml:31]: I currently do not have time to add 191 return annotations but I will start working on it in the future.

ruff might have autofixes for ANN001 in the future https://github.com/astral-sh/ruff/issues/1640

File: .pre-commit-config.yaml
Comment (MarcoGorelli) [inline .pre-commit-config.yaml:29]: shall we add an alias to this, so people don't see `ruff` twice and risk confusion?
something like
```
- id: ruff
  name: ruff-selected-autofixes
  alias: ruff-selected-autofixes
```
?

**Review by MarcoGorelli — APPROVED** [review]
small comment, but rest looks good

File: .pre-commit-config.yaml
Comment (twoertwein) [reply .pre-commit-config.yaml:29]: Okay - could then also re-use it for other autofixes in the future.

**Review by MarcoGorelli — APPROVED** [review]
nice one, never liked having `scripts/run_autotyping`

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @twoertwein

**Pull request merged**
